### PR TITLE
[#880] Fix passive/active collision in rigid body mode

### LINKED ISSUE #880: Passive collision does not work as expected when using rigid body physics
excalibur-passive-collision-error
### Steps to Reproduce
```typescript
var game = new ex.Engine({
    width: 600,
    height: 400
});

ex.Physics.collisionResolutionStrategy = ex.CollisionResolutionStrategy.RigidBody;

var activeBlock = new ex.Actor(200, 200, 50, 50, ex.Color.Red.clone());
activeBlock.collisionType = ex.CollisionType.Active;
activeBlock.vel.x = 100;
game.add(activeBlock);

var passiveBlock = new ex.Actor(400, 200, 50, 50, ex.Color.DarkGray.clone());
passiveBlock.collisionType = ex.CollisionType.Passive;
passiveBlock.vel.x = -100;
game.add(passiveBlock);

game.start();
```
### Expected Result
The dark gray (passive) block should not be affected by the red (active) block.

### Actual Result
The two blocks collide.

Removing `ex.Physics.collisionResolutionStrategy = ex.CollisionResolutionStrategy.RigidBody;` results in the expected behavior.

### Environment
- browsers and versions: Chrome 60.0.3112.113 (Official Build) (64-bit), Microsoft Edge 40.15063.0.0, Mozilla Firefox 55.0.3 (32-bit)
- operating system: Microsoft Windows 10 Pro
- Excalibur versions: 0.12.0, 0.12.1

### PR DESCRIPTION
**Pull request by eonarheim**
## Please merge #892 before this
Closes #880 

## Changes:

- Fix bug where passive/active collisions were resolving when they shoulding
- Added a unit test to confirm and fix the bug
- Added a visual test to confirm that things behave as expected
- Added a new collision event type called "postcollision" which first when resolution happens
- Updated changelog

### CODE REVIEW

File: src/engine/Actor.ts [hunk 1]
```diff
@@ -471,6 +471,8 @@ export class Actor extends Class implements IActionable, IEvented {
       }
    }
 
+   public on(eventName: Events.collision, handler: (event?: CollisionEvent) => void): void;
```
Comment (kamranayub) [inline src/engine/Actor.ts]: Let's rename this to precollision and update references (deprecate `collision`)

File: src/engine/Actor.ts [hunk 2]
```diff
@@ -4,7 +4,7 @@ import { BoundingBox } from './Collision/BoundingBox';
 import { Texture } from './Resources/Texture';
 import {
    InitializeEvent, KillEvent, PreUpdateEvent, PostUpdateEvent,
-   PreDrawEvent, PostDrawEvent, PreDebugDrawEvent, PostDebugDrawEvent, GameEvent
+   PreDrawEvent, PostDrawEvent, PreDebugDrawEvent, PostDebugDrawEvent, GameEvent, CollisionEvent, PostCollisionEvent
```
Comment (kamranayub) [inline src/engine/Actor.ts]: Rename to `PreCollisionEvent`

File: src/engine/Events.ts [hunk 3]
```diff
@@ -24,6 +24,7 @@ export type preframe = 'preframe';
 export type postframe = 'postframe';
 
 export type collision = 'collision';
```
Comment (kamranayub) [inline src/engine/Events.ts]: Rename to `precollision`

File: src/engine/Actor.ts (hunk 1 above)
Comment (eonarheim) [reply src/engine/Actor.ts]: Ah in actor `update` and `postupdate` are equivalent with this todo

```typescript
      // TODO: Obsolete `update` event on Actor
      this.eventDispatcher.emit('update', new PostUpdateEvent(engine, delta, this));
      this.emit('postupdate', new PostUpdateEvent(engine, delta, this));

```

**Pull request merged**